Sources/Internal/Communicator.swift

```diff
@@ -109,8 +109,9 @@ class Communicator {
 
         func addOrUpdate(_ session: Session) {
             dispatchPrecondition(condition: .notOnQueue(queue))
+            let key = session.url.absoluteString
             queue.sync { [weak self] in
-                self?.sessions[session.url] = session
+                self?.sessions[key] = session
```

Comment:
@sxiaojian88 Could you please explain this a bit more because I don't understand why is the key needed here and session is not working?

session is a struct, as well as WCURL. I assume that when the closure is executed, the session is captured, so there'll be a copy of the session. 

Also, the closure is synchronous, meaning that there'll be no change to session from the line 112 to the line 114.

Reply:
We met a crash in here, so I replace `struct` with `string` then crash disappeared. I don't know the reason.
and I don't know if it will happen to someone else.

Reply:
ok, got it